### ISSUE TITLE
PHP deprecated warnings 

### ISSUE THREAD

**Issue by westonruter** [Automattic/amp-wp#7604]
### Bug Description

I noticed on the [AMP report on plugintests.com](https://plugintests.com/plugins/wporg/amp/latest) that there are 5 deprecation warnings being reported:

```
[29-Aug-2023 18:22:43 UTC] PHP Deprecated:  strstr(): Passing null to parameter #1 ($haystack) of type string is deprecated in /wp-includes/functions.php on line 1144
[29-Aug-2023 18:22:43 UTC] PHP Deprecated:  stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /wp-includes/functions.php on line 1151
[29-Aug-2023 18:22:43 UTC] PHP Deprecated:  stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /wp-includes/functions.php on line 1154
[29-Aug-2023 18:22:43 UTC] PHP Deprecated:  str_contains(): Passing null to parameter #1 ($haystack) of type string is deprecated in /wp-includes/functions.php on line 1161
[29-Aug-2023 18:22:43 UTC] PHP Deprecated:  str_contains(): Passing null to parameter #1 ($haystack) of type string is deprecated in /wp-includes/functions.php on line 1164
```

It's not clear to me whether core is actually at fault or not, as the stack traces don't indicate how the AMP plugin is involved. But it would be good to investigate.

### Expected Behaviour

No deprecation warnings should be reported for the plugin.

### Plugin Version

latest

### AMP plugin template mode

Standard, Transitional, Reader

**Comment by thelovekesh** [comment]
@westonruter Reported deprecation warnings are caused by the AMP plugin. Even in the PHPUnit tests CI, we thought all deprecation warnings were due to the GB plugin, but they were due to the AMP plugin as well.

https://github.com/ampproject/amp-wp/blob/b7a1e1fe395494dca2ed90bf54dc6542aede0e7c/includes/validation/class-amp-validation-error-taxonomy.php#L2889-L2892

`get_edit_post_link( $post_id, 'raw' )` always returns null, and later these deprecation warnings are generated within `add_query_arg()`. Seems like at this point of execution user does not have permission to edit the post hence `get_edit_post_link( $post_id, 'raw' )`  is returning null.

**EDIT:**
**The above one is only responsible for deprecation warnings in unit tests.**

**Comment by thelovekesh** [comment]
The deprecation warnings reported by `plugintests` are visible on the `AMP > Error Index` page with PHP 8.1. 

<details><summary>Deprecation Warnings</summary>


![image](https://github.com/ampproject/amp-wp/assets/54371619/1a7e308f-28c9-4d6b-af18-7d1a262e9df0)



</details> 

<details><summary>Stack trace</summary>

```
Array
(
    [0] => Array
        (
            [file] => /app/public/core-dev/build/wp-admin/includes/class-wp-list-table.php
            [line] => 1782
            [function] => handle_row_actions
            [class] => WP_Terms_List_Table
            [type] => ->
        )

    [1] => Array
        (
            [file] => /app/public/core-dev/build/wp-admin/includes/class-wp-terms-list-table.php
            [line] => 352
            [function] => single_row_columns
            [class] => WP_List_Table
            [type] => ->
        )

    [2] => Array
        (
            [file] => /app/public/core-dev/build/wp-admin/includes/class-wp-terms-list-table.php
            [line] => 260
            [function] => single_row
            [class] => WP_Terms_List_Table
            [type] => ->
        )

    [3] => Array
        (
            [file] => /app/public/core-dev/build/wp-admin/includes/class-wp-list-table.php
            [line] => 1618
            [function] => display_rows_or_placeholder
            [class] => WP_Terms_List_Table
            [type] => ->
        )

    [4] => Array
        (
            [file] => /app/public/core-dev/build/wp-admin/edit-tags.php
            [line] => 611
            [function] => display
            [class] => WP_List_Table
            [type] => ->
        )

)
```

</details> 

Inside `handle_row_actions` it generates an [edit link](https://github.com/WordPress/wordpress-develop/blob/9a87bc5d4e1c77d58bf859523c55a62b5ae651f5/src/wp-admin/includes/class-wp-terms-list-table.php#L472-L476) for a given term ID which then checks if the current user can edit a particular term ID with edit_term capability. In our case we are using custom capabilities hence it passes `null` to the `add_query_arg()` which then generates deprecation warnings.

**Comment by pavanpatil1** [comment]
QA Passed ✅ 


Cross-verified the issue and it is working fine. With the WordPress 6.3 the warnings are still visible. These warnings have been successfully addressed in WordPress version 6.4 through this Pull Request - https://github.com/WordPress/wordpress-develop/pull/5198



Before fix        |  After fix
:----------------:|:-------------------------:
![image](https://github.com/ampproject/amp-wp/assets/44057535/dd071129-060c-4894-9d76-b8876ace381b)|![image](https://github.com/ampproject/amp-wp/assets/44057535/7eb97c75-1a77-4d6d-b44f-e42450650c1a) |
